test/types/dependencies.test-d.ts

```diff
@@ -0,0 +1,49 @@
+import { expectError, expectType } from 'tsd'
+import { createPlugin } from './registerV2.test-d'
+import fastify from '../../fastify'
+
+export const plugin1 = createPlugin((instance) =>
+  instance
+    .decorateReply('testPropSync')
+    .decorateReply('testValueSync', 'testValue')
+    .decorateReply('testFnSync', () => 12345))
```

Comment:
Could you give an example of how a plugin like [fastify-static](https://github.com/fastify/fastify-static) could be turned into a viable dependencies?

Reply:
The goal in improving typing is also regarding the existing.

Reply:
Ok, i will do that in fastify-static.